beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -231,8 +232,10 @@ pub struct BeaconChain<T: BeaconChainTypes> {
     pub(crate) head_tracker: Arc<HeadTracker>,
     /// A cache dedicated to block processing.
     pub(crate) snapshot_cache: TimeoutRwLock<SnapshotCache<T::EthSpec>>,
-    /// Caches the shuffling for a given epoch and state root.
+    /// Caches the attester shuffling for a given epoch and shuffling keying root.
     pub(crate) shuffling_cache: TimeoutRwLock<ShufflingCache>,
+    /// Caches the beacon block proposer shuffling for a given epoch and shuffling keying root.
+    pub(crate) beacon_proposer_cache: RwLock<BeaconProposerCache>,
```

Comment:
Seeing as this is never `.read()` I think it may as well be a `Mutex`, but I don't suppose it matters too much